Source/BSN.Commons/Infrastructure/MessageBroker/Kafka/KafkaEventAggregator.cs

```diff
@@ -0,0 +1,192 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using Confluent.Kafka;
+using System.Threading.Tasks;
+using BSN.Commons.Infrastructure.MessageBroker.EventAggregator;
+using BSN.Commons.Infrastructure.MessageBroker.EventContracts.EventAggregator;
+using BSN.Commons.Infrastructure.MessageBroker.EventContracts.EventAggregator.Contracts;
+using BSN.Commons.Infrastructure.MessageBroker.EventContracts.EventAggregator.EventModels;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace BSN.Commons.Infrastructure.MessageBroker.Kafka
+{
+    /// <summary>
+    /// Represents an implementation of an event aggregator,
+    /// a fundamental component in an event-driven architecture,
+    /// responsible for aggregating events and broadcasting them to interested subscribers.
+    /// this implementation uses Kafka as the underlying message broker.
+    /// </summary>
+    public class KafkaEventAggregator : IEventAggregator
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="KafkaEventAggregator"/> class.
+        /// </summary>
+        /// <param name="connectionOptions">The Kafka connection options.</param>
+        /// <param name="subscriptionManager">The subscription manager that manages event subscriptions.</param>
+        /// <param name="logger">The logger.</param>
+        public KafkaEventAggregator(
```

Comment:
I suggest add another constructor to accept consumer and producer Kafka factory and use it

Reply:
done